src/layers/embeddings/qk_rotary_embeddings.rs

```diff
@@ -1,8 +1,98 @@
-use candle_core::{Device, IndexOp, Tensor};
+use candle_core::{IndexOp, Tensor};
+use candle_nn::VarBuilder;
 use snafu::{ensure, ResultExt, Snafu};
 
 use crate::kv_cache::KeyValueCache;
-use crate::layers::embeddings::{RotaryEmbeddings, RotaryEmbeddingsError};
+use crate::layers::embeddings::{RotaryEmbeddings, RotaryEmbeddingsConfig, RotaryEmbeddingsError};
+
+/// Configuration for query-key rotary embeddings.
+#[derive(Debug)]
```

Comment:
```suggestion
#[derive(Debug, Clone)]
```
Just adding clone wherever it makes sense for the configs.